src/FollowMe/FollowMe.cc

```diff
@@ -0,0 +1,178 @@
+    _locationInfo->setUpdateInterval(_locationInfo->minimumUpdateInterval());
+    connect(_locationInfo, SIGNAL(positionUpdated(QGeoPositionInfo)), this, SLOT(_setGPSLocation(QGeoPositionInfo)));
+
+    // set up the mavlink motion report timer
+
+    _gcsMotionReportTimer.setInterval(REPORT_RATE);
+    _gcsMotionReportTimer.setSingleShot(false);
+    connect(&_gcsMotionReportTimer, &QTimer::timeout, this, &FollowMe::_sendGCSMotionReport);
+
+    enable();
+}
+
+FollowMe::~FollowMe()
+{
+    disable();
+}
+
+void FollowMe::enable()
+{
+    _locationInfo->startUpdates();
+    _gcsMotionReportTimer.start();
+}
+
+void FollowMe::disable()
+{
+    _locationInfo->stopUpdates();
+    _gcsMotionReportTimer.stop();
+}
+
+void FollowMe::_setGPSLocation(QGeoPositionInfo geoPositionInfo)
+{
+    if (geoPositionInfo.isValid())
+    {
+        // get the current location coordinates
+
+        QGeoCoordinate geoCoordinate = geoPositionInfo.coordinate();
+
+        if(_simulate_motion == true) {
+            _motionReport.lat_int = 47.3977420*1e7;
+            _motionReport.lon_int = 8.5455941*1e7;
+            _motionReport.alt = 488.00;
+        } else {
+            _motionReport.lat_int = geoCoordinate.latitude()*1e7;
+            _motionReport.lon_int = geoCoordinate.longitude()*1e7;
+            _motionReport.alt = geoCoordinate.altitude();
+        }
+
+        // get the current eph and epv
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::HorizontalAccuracy) == true) {
+            _motionReport.pos_std_dev[0] = _motionReport.pos_std_dev[1] = geoPositionInfo.attribute(QGeoPositionInfo::HorizontalAccuracy);
+        }
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalAccuracy) == true) {
+            _motionReport.pos_std_dev[2] = geoPositionInfo.attribute(QGeoPositionInfo::VerticalAccuracy);
+        }
+
+        // calculate z velocity if it's availible
+
+        if(geoPositionInfo.hasAttribute(QGeoPositionInfo::VerticalSpeed)) {
+            _motionReport.vz = geoPositionInfo.attribute(QGeoPositionInfo::VerticalSpeed);
+        }
+
+        // calculate x,y velocity if it's availible
+
+        if((geoPositionInfo.hasAttribute(QGeoPositionInfo::Direction)   == true) &&
+           (geoPositionInfo.hasAttribute(QGeoPositionInfo::GroundSpeed) == true)) {
+
+            qreal direction = _degreesToRadian(geoPositionInfo.attribute(QGeoPositionInfo::Direction));
+            qreal velocity = geoPositionInfo.attribute(QGeoPositionInfo::GroundSpeed);
+
+            _motionReport.vx = cos(direction)*velocity;
+            _motionReport.vy = sin(direction)*velocity;
+        }
+    }
+}
+
+void FollowMe::_sendGCSMotionReport(void)
+{
+    QmlObjectListModel & vehicles = *_toolbox->multiVehicleManager()->vehicles();
+    MAVLinkProtocol* mavlinkProtocol = _toolbox->mavlinkProtocol();
+    mavlink_follow_target_t follow_target;
+
+    memset(&follow_target, 0, sizeof(mavlink_follow_target_t));
```

Comment:
Yeah, I will go and add these. There are a bunch that I didn't add because I wasn't using them. I won't be able to add all of them yet, but hopefully I'll eventually I get around to all of them!